src/components/Timeline/TimelineEvent.tsx

```diff
@@ -108,345 +118,351 @@ const GenericTimelineEvent = ({
-            >
-              {formatAud(event.amount)}
-            </Tooltip>
-            {TICKER}
-          </>
-        )
-      }
-    }
-
-    const header = headersMap[stage][userType]
-    const title = (
-      <span className={styles.titleContainer}>{bodyMap[stage][userType]}</span>
-    )
+  const renderCancelledSent = () => {
     return (
-      <GenericTimelineEvent
-        onClick={onClick}
-        className={className}
-        header={header}
-        title={title}
-        blockNumber={event.blockNumber}
-      />
+      <>
+        {`Cancelled request to decrease delegation to`}
+        <span className={styles.titleSpacingLeft}>
+          {formatShortWallet(event.serviceProvider)}
+        </span>
+        <span className={styles.titleSpacingLeft}>{'by'}</span>
+        <Tooltip
+          position={Position.TOP}
+          text={formatWei(event.amount)}
+          className={clsx(styles.titleSpacingLeft, styles.titleSpacingRight)}
+        >
+          {formatAud(event.amount)}
+        </Tooltip>
+        {TICKER}
+      </>
     )
   }
 
-  if ('registrationAction' in event) {
-    // did it register or deregister?
-    const didRegister = event.registrationAction === 'register'
-
-    // is it discovery-node or creator-node
-    const isDiscovery =
-      event.serviceType in ['discovery-node', 'discovery-provider']
-    const onClick = () => {
-      if (parentOnClick) parentOnClick()
-      const route = isDiscovery
-        ? discoveryNodePage(event.spID)
-        : contentNodePage(event.spID)
-      pushRoute(route)
-    }
-    const header = didRegister ? 'REGISTERED SERVICE' : 'DEREGISTERED SERVICE'
-    const title = (
-      <span className={styles.titleContainer}>
-        {`${didRegister ? 'Registered' : 'Deregistered'} ${
-          event.serviceType
-        } at ${event.endpoint}`}
-      </span>
-    )
+  const renderCancelledReceived = () => {
     return (
-      <GenericTimelineEvent
-        onClick={onClick}
-        className={className}
-        header={header}
-        title={title}
-        blockNumber={event.blockNumber}
-      />
+      <>
+        {`Delegator ${formatShortWallet(
+          event.delegator
+        )} cancelled request to decrease delegation by`}
+        <Tooltip
+          position={Position.TOP}
+          text={formatWei(event.amount)}
+          className={clsx(styles.titleSpacingLeft, styles.titleSpacingRight)}
+        >
+          {formatAud(event.amount)}
+        </Tooltip>
+        {TICKER}
+      </>
     )
   }
 
-  // SPs: staking increase + decrease events
-  // stake actions can be
-  // increases, decreases requested/evaluated/(eventually cancelled)
-  if ('stakeAction' in event) {
-    const action:
-      | 'increase'
-      | 'decreaseRequested'
-      | 'decreaseEvaluated'
-      | 'decreaseCancelled' = event.stakeAction
-
-    const onClick = () => {
-      if (parentOnClick) parentOnClick()
-      // do nothing for stake actions
+  const headersMap = {
```

Comment:
define outside of component probs

Reply:
normally would agree but in this case since `bodyMap` has to be defined internally, I think it feels best to have `headerMap` colocated with `bodyMap`